cmd/kube-apiserver/app/server.go

```diff
@@ -156,10 +154,6 @@ func (s *APIServer) AddFlags(fs *pflag.FlagSet) {
 		"will be used. If --bind-address is unspecified, the host's default interface will "+
 		"be used.")
 	fs.Var(&s.BindAddress, "public-address-override", "DEPRECATED: see --bind-address instead")
-	fs.IntVar(&s.ReadOnlyPort, "read-only-port", s.ReadOnlyPort, ""+
```

Comment:
Removing ReadOnlyPort seems to me to be the most likely to break people's existing clusters.  Do you want to put it in a separate PR so it can be rolled back independent of the other changes?